kernel/src/process/process/mod.rs

```diff
@@ -225,7 +225,10 @@ impl Process {
             prof_clock,
         })
     }
-
+    pub fn envp(&self) -> Result<Vec<CString>> {
+        let init_stack_reader = self.init_stack_reader();
+        init_stack_reader.envp()
```

Comment:
I don't think that you need to add a new `envp` method. The user of `Process` can invoke the `init_stack_reader` method first and then call the `envp` method herself.